src/server/services/csvPipeline/uploadMeters.js

```diff
@@ -41,21 +42,71 @@ async function uploadMeters(req, res, filepath, conn) {
 	try {
 		for (let i = 0; i < meters.length; i++) {
 			let meter = meters[i];
+			//validation for boolean values
+			validateBooleanFields(meter, i);
+
+			// Validate min and max values
+			validateMinMaxValues(meter, i);
+
 			// First verify GPS is okay
 			// This assumes that the sixth column is the GPS as order is assumed for now in a GPS file.
 			const gpsInput = meter[6];
 			// Skip if undefined.
 			if (gpsInput) {
 				// Verify GPS is okay values
 				if (!isValidGPSInput(gpsInput)) {
-					let msg = `For meter ${meter[0]} the gps coordinates of ${gpsInput} are invalid`;
+					let msg = `For meter ${meter[0]} the gps coordinates of ${gpsInput} are invalid.`;
 					throw new CSVPipelineError(msg, undefined, 500);
 				}
 				// Need to reverse latitude & longitude because standard GPS gives in that order but a GPSPoint for the
 				// DB is longitude, latitude.
 				meter[6] = switchGPS(gpsInput);
 			}
 
+			// Verify area unit - have to parse float to ensure it can read in integer numbers like "33"
+			const areaInput = parseFloat(meter[9]);
+			if (areaInput) {
```

Comment:
This check can fail in certain circumstances. I tested and 0 (zero) and -1 returns false so the area check does not happen. I also notice that 0.1 or any value with a decimal part does not work but it should. Finally, this allowed letters for numbers to pass the test. This may impact other tests you added. I did not try a fix so let me know if you have issues in resolving.